src/abi/mod.rs

```diff
@@ -27,6 +27,18 @@ pub fn generate_abi(
 
             (serde_json::to_string_pretty(&abi).unwrap(), "contract")
         }
+        Target::Solana => {
+            if verbose {
+                eprintln!(
+                    "info: Generating Anchor metadata for contract {}",
+                    ns.contracts[contract_no].name
+                );
+            }
+
+            let idl = anchor::generate_anchor_idl(contract_no, ns);
+
+            (serde_json::to_string_pretty(&idl).unwrap(), "json")
+        }
         _ => {
             if verbose {
                 eprintln!(
```

Comment:
Should we remove the Ethereum ABI generation?

Reply:
It is true this is dead code for now. We have other incomplete evm code too. I really hope we can have an evm code emitter in solang one day, this would capture a huge market. I'd rather leave it in place to show "another piece of evm code emitting that we have in place".